src/ColorPickerBubbleMenu/index.tsx

```diff
@@ -0,0 +1,145 @@
+ * `ColorPickerBubbleMenu` as a child of the component where you call `useEditor` and
+ * render your `RichTextField` or `RichTextContent`. (The bubble menu itself
+ * will be positioned appropriately no matter where you put it in your React
+ * tree, as long as it is re-rendered whenever the Tiptap `editor` forces an
+ * update, which will happen if it's a child of the component using
+ * `useEditor`).
+ */
+export default function ColorPickerBubbleMenu({
+  swatchColors,
+  defaultTextColor = "",
+  ColorPickerProps,
+  ...controlledBubbleMenuProps
+}: ColorPickerBubbleMenuProps) {
+  const editor = useRichTextEditorContext();
+
+  if (!editor?.isEditable) {
+    return null;
+  }
+
+  if (!("colorPickerBubbleMenuHandler" in editor.storage)) {
+    throw new Error(
+      "You must add the ColorPickerBubbleMenuHandler extension to the useEditor `extensions` array in order to use this component!"
+    );
+  }
+  const handlerStorage = editor.storage
+    .colorPickerBubbleMenuHandler as ColorPickerBubbleMenuHandlerStorage;
+
+  // Update the menu step if the bubble menu state has changed
+  const menuState = handlerStorage.state;
+
+  // Determine if all of the selected content shares the same set color.
+  const allCurrentTextStyleAttrs: TextStyleAttrs[] =
+    getAttributesForEachSelected(editor.state, "textStyle");
+  const isTextStyleAppliedToEntireSelection = !!editor.isActive("textStyle");
+  const currentColors: string[] = allCurrentTextStyleAttrs.map(
+    // Treat any null/missing color as the default color
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    (attrs) => attrs.color || defaultTextColor
+  );
+  if (!isTextStyleAppliedToEntireSelection) {
+    // If there is some selected content that does not have textStyle, we can
+    // treat it the same as a selected textStyle mark with color set to the
+    // default
+    currentColors.push(defaultTextColor);
+  }
+  const numUniqueCurrentColors = new Set(currentColors).size;
+
+  let currentColor;
+  if (numUniqueCurrentColors === 1) {
+    // There's exactly one color in the selected content, so show that
+    currentColor = currentColors[0];
+  } else if (numUniqueCurrentColors > 1) {
+    // There are multiple colors (either explicitly, or because some of the
+    // selection has a color set and some does not and is using the default
+    // color). Similar to other rich text editors like Google Docs, we'll treat
+    // this as "unset" and not show a color indicator in the button or a
+    // "current" color when interacting with the color picker.
+    currentColor = "";
+  } else {
+    // Since no color was set anywhere in the selected content, we should show
+    // the default color
+    currentColor = defaultTextColor;
+  }
+
+  return (
+    <ControlledBubbleMenu
+      editor={editor}
+      open={menuState}
+      {...handlerStorage.bubbleMenuOptions}
+      {...controlledBubbleMenuProps}
+    >
+      <ColorPickerPopperBody
```

Comment:
This new version of the picker doesn't render the same way (no `width` applied like the `ColorPickerPopper` did before) so the UI is changing here and looks a bit rough, particularly for the buttons at the bottom (though I do appreciate making it more compact if it's going to be rendered inline).

Before
<img width="272" alt="Screenshot 2024-05-27 at 10 48 06 AM" src="https://github.com/sjdemartini/mui-tiptap/assets/1647130/1a8f161f-799f-48e6-9db7-d590070a3461">

After
<img width="353" alt="Screenshot 2024-05-27 at 10 41 37 AM" src="https://github.com/sjdemartini/mui-tiptap/assets/1647130/d9672cc1-d416-4a82-8760-3c77b07a9e67">